### ISSUE TITLE
Injecting answers/values for prompts

### ISSUE THREAD

**Issue by faustbrian** [enquirer/enquirer#234]
I am currently dabbling with enquirer and so far it seems solid but how to interact with prompts in tests is quite unclear.

With https://github.com/terkelg/prompts it is possible to inject your response before a prompt is called anywhere. This means that you don't need to have access to the prompt object itself and makes testing anything a breeze. This is important for when you test things like CLI commands where you want to give different answers and act like a user without being able to enter something physically with your keyboard.

All of the enquirer internal tests obviously have access to the prompts directly so you don't need anything like that to test the prompts itself but when you are testing something that uses prompts and you are unable to physically input an answer you'll need either injection or another kind of mechanism to let enquirer know what should be used as a response unless you want to litter your tests with mocks.

The README mentions the following: `All prompts are well-tested, and tests are easy to create without having to use brittle, hacky solutions to spy on prompts or "inject" values.`. Since there is no need for hacky solutions or injection, how do I tell enquirer what value should be used for a specific prompt?

**Comment by faustbrian** [comment]
@doowb @jonschlinkert based on https://github.com/poppinss/prompts I guess it isn't possible to test enquirer prompts without having direct access to them?

**Comment by jonschlinkert** [comment]
This seems like a really strange conclusion given how many unit tests Enquirer has and how easy it is to pass values versus the hacks used in the prompts library. I can only assume this is a disingenuous fanboy issue. Closing.

**Comment by faustbrian** [comment]
Closing due to lack of response.

**Comment by aSapien** [comment]
Quite a shame though. I can't run E2E tests because I don't have direct access to the prompt instance.

**Comment by faustbrian** [comment]
> Quite a bummer though. I can't run E2E tests because I don't have direct access to the prompt instance.

If you can, take a look at https://github.com/terkelg/prompts. The migration from it isn't that difficult and quite quick if you don't use any of the more involved prompts that enquirer offers and testing is a breeze.

**Comment by aSapien** [comment]
Thank you so much! It works perfectly

**Comment by jonschlinkert** [comment]
You’re both saying you didn’t even bother to look at the readme or unit tests. That’s a shame.

**Comment by jonschlinkert** [comment]
@faustbrian there are quite literally hundreds of unit tests in Enquirer that do exactly what you're describing. This is described in the readme. I deleted other comments on this thread because of the unnecessary negativity and off-topic discussion.

> unless you want to litter your tests with mocks

It's far easier to pass data to Enquirer prompts, no need for mocks. I'm not sure what you're missing, but this is one of the problems we solved in Enquirer. What the prompts library does is no different than passing a global object to all prompts, and that library has very few unit tests comparatively, probably because it's a pain to pass data to prompts.
